lib/cylc/scheduler.py

```diff
@@ -1351,20 +1374,223 @@ def suite_shutdown(self):
+            ('stop_point', (self.stop_point and
+                            self.stop_point != self.final_point)),
+            # ^ https://github.com/cylc/cylc/issues/2799#issuecomment-436720805
+            ('stop_task', self.stop_task)
+        ] if value]
+
+        # Check whether there is currently an available host to restart on.
+        try:
+            HostAppointer(cached=False).appoint_host()
+        except EmptyHostList:
+            ret.append('No alternative host to restart suite on.')
+        except Exception:
+            # Any unexpected error in host selection shouldn't be able to take
+            # down the suite.
+            ret.append('Error in host selection:\n' + traceback.format_exc())
+
+        if ret:
+            LOG.critical('Suite cannot automatically restart because:\n' +
+                         '\n'.join(ret))
+            return False
+        return True
+
     def suite_health_check(self, has_changes):
-        """Health check.
+        """Detect issues with the suite or its environment and act accordingly.
+
+        Check if:
+
+        1. Suite is stalled?
+        2. Suite host is condemned?
+        3. Suite run directory still there?
+        4. Suite contact file has the right info?
 
-        Suite run directory still there.
-        Suite contact file has the right info.
         """
+        LOG.debug('Performing suite health check')
+
+        # 1. check if suite is stalled - if so call handler if defined
         if self.stop_mode is None and not has_changes:
             self.check_suite_stalled()
-        now = time()
 
-        if self.time_next_fs_check is None or now > self.time_next_fs_check:
+        now = time()
+        if (self.time_next_health_check is None or
+                now > self.time_next_health_check):
+
+            # 2. check if suite host is condemned - if so auto restart.
+            if self.stop_mode is None:
+                current_glbl_cfg = glbl_cfg(cached=False)
+                for host in current_glbl_cfg.get(['suite servers',
+                                                  'condemned hosts']):
+                    if host.endswith('!'):
+                        # host ends in an `!` -> force shutdown mode
+                        mode = self.AUTO_STOP_RESTART_FORCE
+                        host = host[:-1]
+                    else:
+                        # normal mode (stop and restart the suite)
+                        mode = self.AUTO_STOP_RESTART_NORMAL
+                        if self.auto_restart_time is not None:
+                            # suite is already scheduled to stop-restart only
+                            # AUTO_STOP_RESTART_FORCE can override this.
+                            continue
+
+                    if get_fqdn_by_host(host) == self.host:
+                        # this host is condemned, take the appropriate action
+                        LOG.info('The Cylc suite host will soon become '
```

Comment:
The host is logged in the header of the `log` file so the information is there, however, perhaps it would be helpful to users to stick it in here too?

Reply:
Oh, if that's logged elsewhere that's all good for me, unless you think it's worth logging it here (not enough experience troubleshooting logs to know whether it'd be really useful)